### PR TITLE
Upgraded @actions/cache to 2.0.5

### PR DESCRIPTION
**Pull request by kotewar**
As part of fix needed for #624 , [actions/toolkit](https://github.com/actions/toolkit) was modified recently to warn the user telling no cache is being saved when there are no paths matching the given input paths. 

This PR is to update the @actions/cache dependency with recently published [v2.0.5](https://www.npmjs.com/package/@actions/cache/v/2.0.5) version of the same.

### CODE REVIEW

**Review by tiwarishub — APPROVED** [review]
LGTM

File: RELEASES.md [hunk 1]
```diff
@@ -6,3 +6,9 @@
 ### 3.0.1
 - Added support for caching from GHES 3.5.
 - Fixed download issue for files > 2GB during restore.
+
+### 3.0.2
+- Added support for dynamic cache size cap on GHES.
+
+### 3.0.3
```
Comment (vsvipul) [inline RELEASES.md:13]: why are we doing 2 releases?

Comment (kotewar) [reply RELEASES.md:13]: We're doing only one release i.e. 3.0.3. We had missed updating release notes for the previous release, hence added that information here just to track the same.

Comment (vsvipul) [reply RELEASES.md:13]: Ah nice. you might also want to add these to README.md. I'll suggest removing v2 from there now as they are quite old.

Comment (kotewar) [reply RELEASES.md:13]: Alright, let me do that as well. Will also update the link for 

> Refer [here](https://github.com/actions/cache/blob/v1/README.md) for previous versions 


to [v2](https://github.com/actions/cache/blob/v2/README.md) as it is currently pointing to [v1 README](https://github.com/actions/cache/blob/v1/README.md).

**Comment by kotewar** [comment]
Tested the changes in my local repository. Things are working as expected. Merging the PR.

**Comment by abirhasan132** [comment]
#808

**Review by abirhasan132 — COMMENTED** [review]
8b54d92f410048d51683f5f298411de309e50500

**Pull request merged**